test/generators/php/presets/__snapshots__/JsonSerializablePreset.spec.ts.snap

```diff
@@ -0,0 +1,46 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`PHP_JSON_SERIALIZABLE_PRESET should render jsonSerialize method for class 1`] = `
+"final class Clazz implements JsonSerializable
+{
+  private ?string $prop;
+  private ?string $propMinusWithMinusDash;
+  private mixed $additionalProperties;
+
+  public function getProp(): ?string { return $this->prop; }
+  public function setProp(?string $prop): void { $this->prop = $prop; }
+
+  public function getPropMinusWithMinusDash(): ?string { return $this->propMinusWithMinusDash; }
+  public function setPropMinusWithMinusDash(?string $propMinusWithMinusDash): void { $this->propMinusWithMinusDash = $propMinusWithMinusDash; }
+
+  public function getAdditionalProperties(): mixed { return $this->additionalProperties; }
+  public function setAdditionalProperties(mixed $additionalProperties): void { $this->additionalProperties = $additionalProperties; }
+
+  public function jsonSerialize(): array
+  {
+    return [
+      'prop' => $this->prop,
+      'prop-with-dash' => $this->propMinusWithMinusDash,
+      ...$this->additionalProperties,
+    ];
+  }
+}
+"
+`;
+
+exports[`PHP_JSON_SERIALIZABLE_PRESET should render jsonSerialize method for enum 1`] = `
+"enum Enumm implements JsonSerializable
+{
+  case VALUE_MINUS_A;
+  case VALUE_MINUS_B;
+
+  public function jsonSerialize(): mixed
+  {
+    return match($this) {
+      self::VALUE_MINUS_A => \\"value-A\\",
+      self::VALUE_MINUS_B => \\"value-B\\",
+    };
+  }
```

Comment:
Idk if that's the right approach, but have no other idea how to handle not backend enums.

also I guess we should add a test with backed enums. I will add an issue for that

Reply:
How can you define a non-backed enum in JSON schema? If you can provide an example, I could test it.

I added at least a test case for mixed enums: 048c9e17d7e0431f032f707e0810dc93ffd4b3c5

Reply:
That's a good question. Ideally, we should only have backed enums because the protocol needs values.

PHP examples would be:
```php
enum Suit: string
{
    case Hearts = 'H';
    case Diamonds = 'D';
    case Clubs = 'C';
    case Spades = 'S';
}
```

```
enum Week: int
{
    case Monday = 1;
    case Tuesday = 2;
    case Wednesday = 3;
}
```